### PR TITLE
implement single colon typo check, refers to #60

### PR DESCRIPTION
**Pull request by LiNk-NY**
Tries to make some savings by only calling `BiocManager::available` when there is a detection of `:` in the parsed code. 
See #60 for more details

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,10 @@
+CHANGES IN VERSION 1.25
+-----------------------
+
+NEW FEATURES
+
+    o Check for single colon typos and suggest double colons
```
Comment (mtmorgan) [inline NEWS]: provide a bit more context here? I guess this involves pkg::foo() notation, and not some other use of `:`?

File: R/checks.R [hunk 2]
```diff
@@ -901,6 +901,15 @@ checkCodingPractice <- function(pkgdir, parsedCode, package_name)
             handleMessage(msg, indent=8)
     }
 
+    # pkg:fun...
+    msg_sc <- checkSingleColon(Rdir)
+    if (length(msg_sc)) {
+        handleError(" Single colon use found in PKG:FUN; use double colon '::'")
```
Comment (mtmorgan) [inline R/checks.R]: Not sure about this, but is it 'best practice' never to tell the user what they did wrong, but to tell them what they need to do to get it right? Kind of like not planting the seed of bad ideas in their brains...

File: R/checks.R [hunk 3]
```diff
@@ -998,6 +1007,33 @@ check1toN <- function(Rdir){
...
+        colons <- colons[tokens[colons, "text"] != c("1", "2")]
+        tokens[colons, , drop = FALSE]
+    })
+    colon_pres <- Filter(nrow, colon_pres)
+    availpkgs <-
+        if (length(colon_pres))
+            BiocManager::available()
+        else
+            character(0L)
+    msg_sc <- lapply(names(colon_pres), function(rfile, framelist) {
```
Comment (mtmorgan) [inline R/checks.R:1024]: as 'good practice', add availpkgs to the function signature?

**Review by mtmorgan — APPROVED** [review]
Looks really great! Minor comments.

**Comment by LiNk-NY** [comment]
It looks ready to go ! Thanks

**Comment by Kayla-Morrell** [comment]
Hey @LiNk-NY there seem to be some warning being generated from `checkSingleColon`. There is a package I'm reviewing  (https://github.com/Bioconductor/Contributions/issues/1516) where the build report writing a warning, but not an explicit WARNING from BiocCheck https://github.com/Bioconductor/Contributions/issues/1516#issuecomment-648700382.

I did a little debugging and was able to trace down that there are instances where the developer uses single colon but not for importing a function. For example,

```
for(i in 2:length(trunkPath)){
```
and,
```
nodeRange <- c(Ntip(tree)+1):(length(tree$edge.length)+1)
```

There probably can be some code clean up on the part of the developer (I haven't done a full review yet) but I figured I'd bring it to your attention. The function isn't producing a wrong output, because these examples shouldn't produce a WARNING from the function. But within the function these examples are throwing a warning at the following line. https://github.com/Bioconductor/BiocCheck/blob/e0176f6591c771774f6cb05d4ff8bc1b145ad10e/R/checks.R#L1050 Let me know if you need more detail to help clear this up.

**Comment by LiNk-NY** [comment]
See #98

**Pull request merged**